pact/syn-template.pact

```diff
@@ -365,6 +366,7 @@
           { "balance": 0.0, "guard": receiver-guard }
           { "balance" := receiver-balance, "guard" := existing-guard }
           (enforce (= receiver-guard existing-guard) "Supplied receiver guard must match existing guard.")
+          (enforce-reserved receiver receiver-guard)

```

Comment:
In this change, we explicitly disallow burning tokens by requiring the receiving account to be principal. Can we confirm that this is what we want?